### ISSUE TITLE
Not sure if bug with Octane, or if I'm doing something wrong

### ISSUE THREAD

**Issue by NullVoxPopuli** [crashco/ember-template-component-import#10]
I have this template:
```hbs
{{import Intro from 'emberconf-2019/src/ui/routes/application/-components/section-01-intro'}}

<Slides as |actions|>
  <Intro @openPresenterView={{actions.togglePresenterView}} />
...
```

and get this error:
```
Error: Assertion Failed: Could not find component named "emberconf-2019/src/ui/routes/application/-components/section-01-intro" (no component or template with that name was found) index.js:163
```
But, in the console:
```
> Object.keys(window.requirejs.entries).filter(p => p.includes('section-01'))
Array [ "emberconf-2019/src/ui/routes/application/-components/section-01-intro/component", "emberconf-2019/src/ui/routes/application/-components/section-01-intro/template" ]
```

**Comment by davewasmer** [comment]
@NullVoxPopuli under the hood, this addon just takes the import path and passes it directly to the `{{component` helper (except for relative imports - we do some path manipulation there, but it still ultimately is passed to `{{component`). So my guess is that you'll need to drop the `emberconf-2019` prefix?

**Comment by NullVoxPopuli** [comment]
dropping the app name:
```hbs
{{import Intro from 'src/ui/routes/application/-components/section-01-intro'}}
```
```
Error: Assertion Failed: Could not find component named "src/ui/routes/application/-components/section-01-intro" (no component or template with that name was found)
```

with a relative import, I get this error:
```hbs
{{import Intro from './-components/section-01-intro'}}
```

```
Error: Assertion Failed: Could not find component named "../emberconf-2019/src/ui/routes/application/-components/section-01-intro" (no component or template with that name was found)
```

I guess what's strange is that these paths in the errors _seem_ correct. 
  -> for the relative import, I'm lead to believe the "working directory" is the `src` folder, but I think I'd expect `src/ui/components` if using the `component` helper.

I wonder if something broke in ember-cli recently (using master of 2 hours ago)

idk:

**Comment by davewasmer** [comment]
I looks like you're using this in an MU app? If so, I haven't tested the relative path stuff against MU structure at all. I'd be happy to take a PR that corrects any bugs there, but likely won't have time to tackle that myself for a bit (also - hopefully this addon will be obsoleted in MU apps by actual template imports soon).

You can check if it's this addon vs. ember-cli by simply replacing your absolute import statements with `{{component $ABSOLUTE_PATH}}`. If you get the same error - it's not this addon. If you don't, then it's us.

**Comment by NullVoxPopuli** [comment]
gotchya, yeah, it's def an issue with `component`/ember-cli then. 

I'm using a _fresh_ octane app: 

```bash
npx ember-cli new my-app -b ember-octane-app-blueprint
```

**Comment by lifeart** [comment]
Looks like I have an idea how to solve this issue.
we need to alias component path somehow (using helper, js injection or `<script>` inline);

```js

define.alias('emberclear/src/ui/routes/index/-components/compatibility/feature/template', 'emberclear/src/ui/components/${template-hash}/compatibility/feature/template');
define.alias('emberclear/src/ui/routes/index/-components/compatibility/feature/component', 'emberclear/src/ui/components/${template-hash}/compatibility/feature/component');

```

**Comment by lifeart** [comment]
also, we can do handy transform in runtime

```hbs

import { Foo, Bar } from 'src/ui/components'

--- hbs ---

```

to 

```hbs
{{import Foo from 'src/ui/components/foo'}}
{{import Bar from 'src/ui/components/bar'}}
```

**Comment by lifeart** [comment]
### TLDR:

```hbs

import { Foo }  from 'src/ui/routes/boo/-components/'
	
--- hbs ---

```

will be transformed to 

```hbs

	{{import Foo from "src/ui/routes/boo/-components/foo"}}

```

this will be transformed to 

```hbs
{{#let (component (registerAbsoluteImport "current-file-name.hbs" "src/ui/routes/boo/-components/foo")) as |Foo|}}
	...
{{/let}}
```

`registerAbsoluteImport` will return `sdfdf/foo`, and it can be rewrited like

```hbs
{{#let (component `sdfdf/foo`) as |Foo|}}
	... 
{{/let}}
```

**Comment by davewasmer** [comment]
@lifeart is the goal here simply to get the named import syntax? I.e the `import { Foo }` part?

Also, why the `--- hbs ---`?

I'm hesitant to introduce this much complexity if this is just to get `import { Foo }`, especially since that might confuse some folks (since it _looks_ like ES6 named imports, but doesn't really share the semantics at all).

**Comment by lifeart** [comment]
https://gist.github.com/lifeart/58f74c89812c420bcb3044857732292b

**Comment by lifeart** [comment]
@davewasmer, this is Octane feature (named imports and JS syntax on the top of the template), and other hbs stuff should go after `--- hbs ---` line.

**Comment by NullVoxPopuli** [comment]
caveat though: there isn't yet an RFC for this -- just want to keep that in mind :)

**Comment by knownasilya** [comment]
@davewasmer https://github.com/ef4/rfcs/blob/js-compatible-resolving/text/0000-js-compatible-resolution.md is the pre-RFC

**Comment by lifeart** [comment]
lol, my `--- hbs ---` from `https://github.com/wycats/ember-next-experiments#proposed-9`, can be simply replaced to something more agreed

**Comment by lifeart** [comment]
https://github.com/crashco/ember-template-component-import/pull/12/files

**Comment by knownasilya** [comment]
MU support will not be coming, since that was dropped. Otherwise the new syntax is in v1.2.0

**Comment by NullVoxPopuli** [comment]
:+1:
